### PR TITLE
2.31.12: add in_exec initialization fix and other bug fixes from upstream

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by PettitWesley** [comment]
This was an existing fix upstream which wasn't backported. 

I'm going to look tomorrow into adding more simple fixes from upstream 2.x into our distro. Similar to how 2.31.11 included some extra fix commits I found upstream not related to any issue report. 

The windows pipeline runs slowly due to how many versions we have right now... it'd be best to keep the number of patch releases to what's necessary. 

(Of course, splitting into multiple releases is advantageous if any one patch has a problem, since it may prevent customers from consuming other patches, but our pre-release stability tests should give some safety).

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ This release includes:
 Compared to `2.31.11` this release adds:
 * Bug - Fix incorrect decrementing of s3_key_format $INDEX on Multipart Upload failure [aws-for-fluent-bit:653](https://github.com/aws/aws-for-fluent-bit/issues/653)
 * Bug - `cloudwatch` go plugin: fix utf-8 calculation of payload length to account for invalid unicode bytes that will be replaced with the 3 byte unicode replacement character. This bug can lead to an `InvalidParameterException` from CloudWatch when the payload sent is calculated to be over the limit due to character replacement.  [cloudwatch:330](https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit/pull/330) 
+* Bug - fix SIGSEGV crash issue in exec input due to memory initialization [aws-for-fluent-bit:661](https://github.com/aws/aws-for-fluent-bit/issues/661)
+* Bug - record_accessor/rewrite_tag fix to allow single character rules [fluent-bit:7330](https://github.com/fluent/fluent-bit/issues/7330)
+* Bug - filter_modify: fix memory clean up that can lead to crash [fluent-bit:7368](https://github.com/fluent/fluent-bit/issues/7368)
+
```
Comment (PettitWesley) [inline CHANGELOG.md]: I've only added entries to our changelog if the fixes actually had associated issues with a user reporting and explaining impact.

**Review by DrewZhang13 — APPROVED** [review]
LGTM.

just double check if 2.31.12 is released already or not? As i don't see 2.31.12 in release note here: https://github.com/aws/aws-for-fluent-bit/releases/tag/v2.31.11. But the code change seems already have partial 2.31.12 change log merged.

**Review by Claych — APPROVED** [review]

**Comment by PettitWesley** [comment]
@DrewZhang13 it was previously staged for release and then I decided to add some things to it now. It has not been actually released yet.

**Comment by PettitWesley** [comment]
This covers core fixes since April 1st, which seemed to be around when they stopped backporting key fixes... I possibly should've checked older than that

**Pull request merged**